x/lightclient/keeper/rollback.go

```
@@ -60,26 +61,36 @@ func (k Keeper) RollbackCanonicalClient(ctx sdk.Context, rollappId string, lastV
 // and adding consensus states based on the block descriptors
 // CONTRACT: canonical client is already set, state info exists
 func (k Keeper) ResolveHardFork(ctx sdk.Context, rollappID string) error {
-	clientID, _ := k.GetCanonicalClient(ctx, rollappID) // already checked in the caller
+	clientID, _ := k.GetCanonicalClient(ctx, rollappID)                // already checked in the caller
+	stateinfo, _ := k.rollappKeeper.GetLatestStateInfo(ctx, rollappID) // already checked in the caller
+
+	err := k.UpdateClientFromStateInfo(ctx, clientID, &stateinfo)
+	if err != nil {
+		return errorsmod.Wrap(err, "update client from state info")
+	}
+	return nil
+}
+
+// UpdateClientFromStateInfo sets the consensus state from the state info
+// and sets the metadata for the consensus state
+// uses the latest height of the state info to set the client state
+// CONTRACT: canonical client is already set, state info exists
+func (k Keeper) UpdateClientFromStateInfo(ctx sdk.Context, clientID string, stateInfo *rollapptypes.StateInfo) error {
 	clientStore := k.ibcClientKeeper.ClientStore(ctx, clientID)
 
-	stateinfo, _ := k.rollappKeeper.GetLatestStateInfo(ctx, rollappID) // already checked in the caller
+	bd := stateInfo.GetLatestBlockDescriptor()
+	height := bd.Height
 
-	height := stateinfo.StartHeight
 	// sanity check
-	client := getClientStateTM(clientStore, k.cdc)
-	clientHeight := client.GetLatestHeight().GetRevisionHeight()
-	if height <= clientHeight {
+	clientLatestHeight := getClientStateTM(clientStore, k.cdc).GetLatestHeight().GetRevisionHeight()
+	if height <= clientLatestHeight {
 		return gerrc.ErrInternal.Wrapf("client latest height not less than new latest height: new: %d, client: %d",
-			height, clientHeight,
+			height, clientLatestHeight,
```

You could just return here (no error) and not check on the caller
Less code

---

if `if height <= clientLatestHeight {` not satisfied, I prefer to return error instead of silently ignoring